marimo/_ai/_tools/tools/datasource.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2025 Marimo. All rights reserved.
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Optional
+
+from marimo import _loggers
+from marimo._ai._tools.base import ToolBase
+from marimo._ai._tools.types import SuccessResult
+from marimo._ai._tools.utils.exceptions import ToolExecutionError
+from marimo._data.models import DataTable
+from marimo._server.sessions import Session
+from marimo._types.ids import SessionId
+from marimo._utils.fuzzy_match import compile_regex, is_fuzzy_match
+
+LOGGER = _loggers.marimo_logger()
+
+
+@dataclass
+class GetDatabaseTablesArgs:
+    session_id: SessionId
+    query: Optional[str] = None
+
+
+@dataclass
+class TableDetails:
+    connection: str
+    database: str
+    schema: str
+    table: DataTable
+
+
+@dataclass
+class GetDatabaseTablesOutput(SuccessResult):
+    tables: list[TableDetails] = field(default_factory=list)
```

Comment:
i wonder if it would help to add an example sql: e.g. `_df = mo.sql("SELECT * FROM database.schema.table LIMIT 100"